frontend/src/components/pages/UserActions/ActivityDashBoardBarGraph/utils.js

```diff
@@ -0,0 +1,28 @@
+export function prepareDataForBarGraph({ eventTypes, activityLogDayCounts }) {
+  const userActivityLogs = [];
+
+  activityLogDayCounts.forEach((act) => {
+    eventTypes.forEach((ev) => {
+      if (ev.id === act.eventType) {
+        if (
+          !userActivityLogs.some(
+            (userActivityLog) => userActivityLog.date === act.date
+          )
+        ) {
```

Comment:
There is a simple algorithm that works for these kinds of problems. Your code works, but it's pretty complicated. 

but this kind of thing is nice and easy:

```
const eventNames = {}
eventTypes.forEach(eventType => eventNames[eventType.id] = eventType[name])

const result = {}

activityLogDayCounts.forEach(dayCount=>{
    const { date } = dayCount 
    const eventName = eventNames[eventType]   // just look up the name when you need it

    result[date] = result[date] || {date}   // if there is a value, leave it alone. Otherwise initialise it
    result[date][eventName] = dayCount.total  
})

return Object.values(result) // return an array
```

I haven't tested this code. But see how simple it is?  The best data structure for mapping keys to values is not an array.  It's an object.

If you use the right data structure for the job then the code is way simpler.

This kind of problem comes up a lot. When I see these things I just think "gosh, again?" and then I write the code almost on autopilot.  

Just because you need to loop over an array and return an array doesn't mean you need to hit your whole algorithm with an array hammer. Use the best tool for the job, not the tool you saw recently.